lldb/source/Plugins/Process/gdb-remote/ProcessGDBRemote.cpp

```diff
@@ -1612,6 +1612,22 @@ bool ProcessGDBRemote::CalculateThreadStopInfo(ThreadGDBRemote *thread) {
   return false;
 }
 
+void ProcessGDBRemote::ParseExpeditedRegisters(
+    ExpeditedRegisterMap &expedited_register_map, ThreadSP thread_sp) {
+  ThreadGDBRemote *gdb_thread = static_cast<ThreadGDBRemote *>(thread_sp.get());
+  RegisterContextSP gdb_reg_ctx_sp(gdb_thread->GetRegisterContext());
+
+  for (const auto &pair : expedited_register_map) {
+    StringExtractor reg_value_extractor(pair.second);
+    WritableDataBufferSP buffer_sp(
```

Comment:
Orthogonal to this patch, but does this buffer need to be writable? It looks like it's "necessary" because `PrivateSetRegisterValue` takes a non-const reference to the buffer, but does it actually need to modify it?

Reply:
We're decoding the asciihex register values into a byte buffer, this is the byte buffer.

Reply:
Ah, makes sense. I was focussed on `PrivateSetRegisterValue` and missed the extraction bit. Thanks!